### ISSUE TITLE
A list of issues and enhancements by frenchomatic

### ISSUE THREAD

**Issue by MARQAS** [ahmedkaludi/accelerated-mobile-pages#1013]
**Marqas**
- [x] alt tag on logo
- [x] thumbnails on archive & taxonomies need alt tags

- [x] Analytics – should be an admin option to remove the action completely – google reports errors if the script is left in the head but nothing is defined.
- [x] archive title and archive description should be replaced by genesis headline / description.

- [x] use of genesis document title / meta description for AMP title tag and meta description.

- [x]  links on menu should use the canonical url (i.e. slash on the end in most cases).
- [x] Trailing slash on amp urls is often missing

- [x]  If a new AMP front-page (static) is defined (i.e. can’t use the existing home page because it is widgetized) then the canonical url and amphtml need to align properly on the home page. Also comments link keeps showing on the static ‘home’ page.
- [ ] If you are using a static front page which is widgetized then it would be really nice to allow a featured image to be added just to the AMP version without it showing up on the non-AMP version. If this could be done then 6 is no longer an issue.

- [ ] usemetadata(1) doesn’t appear to work robustly – it starts putting in admin for the publisher.

**Sejiro-Hiko**

- [x]  author / date not appearing on blog posts page.

- [x]  nice to have a setting for the length of excerpts as opposed to making a hardcode setting

- [x]  On posts page the author and date doesn’t show.

**Comment by MohammedKaludi** [comment]
Some other Issues related to Genesis https://github.com/ahmedkaludi/accelerated-mobile-pages/issues/984

**Comment by MARQAS** [comment]
Alt tag have been added for homepage loop, archive, tags and blog as well

**Comment by MARQAS** [comment]
User meta data is working fine on my end, asked him the details regarding it

**Comment by MARQAS** [comment]
Genesis SEO added

**Comment by Dinglebat** [comment]
Looking at version 0.9.57 on test domain - my opinions only

In archive.php and index.php (design 2)

` the_archive_title( '<h3 class="page-title">', '</h3>' );`

This will probably be more consistent with the design of pages and better for SEO if it uses a h1  + same class as pages. 

`the_archive_title( '<h1 class="amp-wp-title">', '</h1>' );`


Also this in features.php

`$metadata['mainEntityOfPage'] = trailingslashit($current_url); // proper URL added`

Without the trailing slash it is not the canonical url as is normally used for mainentityofpage. 

Also it is difficult to give every case but there seems to be cases where the canonical url is not being presented for amphtml or rendered in the structured schema.  I can find them if you wish. Menu links will align between amp and non-amp versions with this in features.php

 `$atts['href'] = trailingslashit(trailingslashit( $atts['href'] ) . AMPFORWP_AMP_QUERY_VAR);`

Hope this helps.

**Comment by Frenchomatic** [comment]
Not sure if it is intended but on design 2, the author and modified time is not showing on archives for me - css is set to display: none;

**Comment by MARQAS** [comment]
Hi @Frenchomatic 

Yes, we have added a display: none to it, because most of the user's don't want it. 
We will add an option for it, till then you can remove that css.

**Comment by Frenchomatic** [comment]
WCAG 2.0 level AA issue - suggestion only for compliance not to link images on archives or blog posts page. Two identical adjacent links is to be avoided.

**Comment by Frenchomatic** [comment]
Test system -  tag pages and categories are not showing after upgrade - pages return headers 200 OK but are blank. Home page and blog posts page show fine, articles and other pages are also fine. Tried deleting all plugins and building the AMP again  - still no luck. 

Will now try to clear the database of any fields left behind after plugin deletion,  upload plugins again and do an import of settings.

**Comment by ahmedkaludi** [comment]
@Frenchomatic Could you please test the Development Branch which is https://github.com/ahmedkaludi/accelerated-mobile-pages/tree/development-0.9.58

We are planning to release this in few hours. We are hoping that this would fix all of the issues.

Please let us know

Regards,
Ahmed

**Comment by Frenchomatic** [comment]
Looking good on my end - tag pages and categories are  now functioning. Basically things are working and don't appear broken. Well done - this has all been a pain for you without doubt.  I will report back on anything I find.

**Comment by Frenchomatic** [comment]
Minor issues: 

1. Leave a comment link appearing on home page - no where to turn it off or stop it except removing code in the plugin.  

2. Genesis SEO -  I presume you are still working on this to pull the meta description and doc title from the genesis page settings. 

3. Canonicial urls on menu links are missing / . 

4. Blurred featured image on home page only - in plugin code frontpage_elements.php  it is set to medium but for me it needs to be large. This is because the AMP home page is created from a completely different page called  /amp-home/ where I add in the AMP content and redirect in the admin panel. What is strange is that  /amp-home/amp/ is fine.  It is not getting this line on that page but it does when the home page is actually rendered. 

$image = wp_get_attachment_image_src( get_post_thumbnail_id( $post_id ), 'medium' ); 

5. For the home page amphtml is pointing to a url which is not canonicalized with a / at the end

**Comment by Frenchomatic** [comment]
I have one future enhancement to recommend that I know users will want because they often ask theme developers how to remove it. On blog posts, it puts in a link to the category in the top right hand corner. Many people have single category blogs and they want to hide the category. It is not serving any puposes. A toggle is needed in the admin panel under "single" so the test to echo it fails - something like  

 ```
if ( $ampforwp_categories && !redux ..... ) : ?>
  	<div class="amp-wp-meta amp-wp-tax-category ampforwp-tax-category">
```

**Comment by Frenchomatic** [comment]
4 web sites all upgraded - all identical in terms of theme and plugins - however one of them doesn't have a blog. It is just a set of pages. On this site it is producing NULL on the Person element of structured data for just the the static home page on that site. I see this

`{"@type":"Person","name":null}`

In fact I can't find where it is setting the structured data on any site for person when it is a page. Only if it is an archive (i.e. features.php 	 - `$structured_data_author=  get_userdata( 1 );`)

**Comment by MARQAS** [comment]
@Frenchomatic 
Check in includes\vendor\amp\includes\class-amp-post-template.php
line number: 172

**Comment by MARQAS** [comment]
@Frenchomatic check this code of comments, if it's working well for you or not

**Comment by MARQAS** [comment]
Regarding the featured image:  of course, it will look blurred on desktop AMP view. 
We have added medium intentionally because it will look good in mobile view, otherwise, it will take more resources.
Let me know your thoughts on it

**Comment by Frenchomatic** [comment]
Feature image - problem is that medium is set to 150 x 150 on my web sites. It is no big deal for me to change that. I could just regenerate thumbs for a better size of say 800px x 800px. Drop the request. This is really down to me. 

The leave comments link on static home page is now gone. Well done. I think we close out that issue.

Working through 9.5.8.1 now - give me 30 mins to report back

**Comment by Frenchomatic** [comment]
Issue of meta description and title on genesis home page now basically resolved. Minor issue. Getting spaces either side of the title  e.g. <title> my tittle </title>

**Comment by Frenchomatic** [comment]
These issues for me are left 

Features.php

These gives the true canonical urls for menu, structured data and canonical in amp pages (home page)

$amp_url =  trailingslashit(trailingslashit($current_archive_url).'amp');

$metadata['mainEntityOfPage'] = trailingslashit($current_url); // proper URL added
    
$atts['href'] = trailingslashit(trailingslashit( $atts['href'] ) . AMPFORWP_AMP_QUERY_VAR);

Then back to the problem above where we get this on home page when there is no blog. 

{"@type":"Person","name":null}

**Comment by Frenchomatic** [comment]
Marqas - I am going to send you an email to team3@ - it will show the problem of the featured image on the home page not being put into the structured data. It is using the default one instead. I have purposefully set a different featured image for the home page to the default so you will see the issue immediately. I will also send you an example of a site where {"@type":"Person","name":null}. There is weirdness in user_data(1) when no blog is being used.

**Comment by Frenchomatic** [comment]
So includes\vendor\amp\includes\class-amp-post-template.php function  is doing something strange. 

Reference your comment about setting the author on pages. 

On home page when there is no blog on the site at all. It goes through the function below twice. The first time the author is NULL and on the second pass it gets the correct author display name. However, the structured data element is being set on the first time round and doesn't get reset with the correct data on the second. 

```
private function build_post_data() {
		$post_title = get_the_title( $this->ID );
		$post_publish_timestamp = get_the_date( 'U', $this->ID );
		$post_modified_timestamp = get_post_modified_time( 'U', false, $this->post );
		$post_author = get_userdata( $this->post->post_author );
**echo "--- why is this null first time round --- " . $post_author->display_name;**
```

And this is what is gets echoed to the screen

why is this null first time round --- --- why is this null first time round ---  mydisplayname

Conversely when there is a blog on the site,  the static home page gives this 

--- why is this null first time round --- mydisplayname--- why is this null first time round --- mydisplayname

**Comment by Frenchomatic** [comment]
Google recommends that within the structured schema.org for an AMP article, a description should be supplied as well as a headline. It isn’t mandatory but it is something they would prefer to see given.

Would it be worth considering setting a structured description to at least take the meta description until a better solution can be found?

**Comment by Frenchomatic** [comment]
As an enhancement down the line -  I think there is a real need to allow json to be embedded into the head of an AMP article via the editor.  To currently do it requires a lot of mucking about in functions.php.  For example, it is currently quite difficult to put a local business structured data type onto an AMP article.

**Comment by MohammedKaludi** [comment]
Thank you @Frenchomatic,  I have created a separate ticket for SD and will sure improve this part in future update.

Mohammed Kaludi

**Comment by MARQAS** [comment]
If you are using a static front page which is widgetized then it would be really nice to allow a featured image to be added just to the AMP version without it showing up on the non-AMP version. If this could be done then 6 is no longer an issue.

This one will be resolved here #1014

**Comment by ahmedkaludi** [comment]
There has been a lot of tasks, Commits, Conversations discussed on this ticket. I am closing this ticket. For further discussions, let's create one issue per ticket.
